lib/btrfs_test.pm

```diff
@@ -40,14 +41,35 @@ way to get DBus-less environment is to enter rescue.target via systemctl.
 =cut
 sub snapper_nodbus_setup {
     my ($self) = @_;
-    if (script_run('! systemctl is-active dbus')) {
-        script_run('systemctl rescue', 0);
-        assert_screen('emergency-shell', 120);
-        type_password;
-        send_key 'ret';
-        $self->set_standard_prompt('root');
-        assert_screen 'root-console';
+    # serial console prints text style as ascii chars in PS1 by default.
+    my $escseq = '[\]\[\(\)\e\d\w\s]*';
+
+    my $rescue_service_config_path   = "/usr/lib/systemd/system/rescue.service";
+    my $rescue_service_override_path = "/etc/systemd/system/rescue.service";
+
+    return unless script_run('! systemctl is-active dbus');
+    if (!serial_terminal::select_virtio_console()) {
+        die 'Selection of virtio console failed.';
     }
+
+    # Replace StandardInput in rescue.service file and add TTYPath.
+    assert_script_run "cp $rescue_service_config_path $rescue_service_override_path";
+    assert_script_run "sed -i 's@^StandardInput=.*\$\@StandardInput=tty\\nTTYPath='\$(tty)'\@g' '$rescue_service_override_path'";
+
+    assert_script_run "systemctl reenable rescue";
+    type_string "systemctl rescue";
+    wait_serial "systemctl rescue";
+    type_string "\n";
+    wait_serial ".*or press Control-D to continue.*:";
+    type_password;
+    type_string "\n";
+
+    wait_serial "$escseq:~ #$escseq";
+    type_string 'PS1="# "; export TERM=dumb; stty cols 2048';
+    wait_serial 'PS1="# "; export TERM=dumb; stty cols 2048';
+    type_string "\n";
+    wait_serial "# ";
+    $self->set_standard_prompt('root');
```

Comment:
This part is repeating some code in lib/serial_terminal.pm. You can refactor that file to add a function which sets the prompt and use it here and in the normal login function for virtio-terminal. You just need to verify that the LTP tests still run after the refactoring.